cloud/openstack/os_user_facts.py

```diff
@@ -0,0 +1,172 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_user_facts
+short_description: Retrieve facts about one or more OpenStack users
+extends_documentation_fragment: openstack
+version_added: "2.1"
+author: "Ricardo Carrillo Cruz (@rcarrillocruz)"
+description:
+    - Retrieve facts about a one or more OpenStack users
+requirements:
+    - "python >= 2.6"
+    - "shade"
+options:
+   user:
+     description:
+        - Name or ID of the user
+     required: true
+   domain:
+     description:
+        - Name or ID of the domain containing the user if the cloud supports domains
+     required: false
+     default: None
+   filters:
+     description:
+        - A dictionary of meta data to use for further filtering.  Elements of
+          this dictionary may be additional dictionaries.
+     required: false
+     default: None
+'''
+
+EXAMPLES = '''
+# Gather facts about previously created users
+- os_user_facts:
+    cloud: awesomecloud
+- debug: var=openstack_users
+
+# Gather facts about a previously created user by name
+- os_user_facts:
+    cloud: awesomecloud
+    name: demouser
+- debug: var=openstack_users
+
+# Gather facts about a previously created user in a specific domain
+- os_user_facts
+    cloud: awesomecloud
+    name: demouser
+    domain: admindomain
+- debug: var=openstack_users
+
+# Gather facts about a previously created user in a specific domain
+  with filter
+- os_user_facts
+    cloud: awesomecloud
+    name: demouser
+    domain: admindomain
+    filters:
+      enabled: False
+- debug: var=openstack_users
+'''
+
+
+RETURN = '''
+openstack_users:
+    description: has all the OpenStack facts about users
+    returned: always, but can be null
+    type: complex
+    contains:
+        id:
+            description: Unique UUID.
+            returned: success
+            type: string
+        name:
+            description: Name given to the user.
+            returned: success
+            type: string
+        description:
+            description: Description of the user
+            returned: success
+            type: string
+        enabled:
+            description: Flag to indicate if the user is enabled
+            returned: success
+            type: bool
+        domain_id:
+            description: Domain ID containing the user
+            returned: success
+            type: string
+        default_project_id:
+            description: Default project ID of the user
+            returned: success
+            type: string
+        email:
+            description: Email of the user
+            returned: success
+            type: string
```

Comment:
you are right, although I'm not sure why it is returned and what is it about.

In the API docs is not listed as response parameter:

http://developer.openstack.org/api-ref-identity-v3.html#users-v3

I will put it on the return string, but will leave it as 'username of the user', don't really know what is supposed to be this value and if this is  a bug on keystone client or maybe the docs?